### PR TITLE
Cleanup for Embeds and its components

### PR DESCRIPTION
**Pull request by skippi**
This PR makes the following changes:

1. Updates the typespecs for Embed fields and its components.
2. Updates the `to_struct/1` functions for each Embed and its components.
3. Implements the Poison.Encoder protocol for each Embed and its components.

### CODE REVIEW

**Comment by skippi** [comment]
UPDATE: We've decided to expand this PR to cover these topics:

* Added builder functions for simplifying the embed building process.
* Added tests to test these builder functions.

File: lib/nostrum/struct/embed.ex [hunk 1]
```diff
@@ -1,6 +1,41 @@
...
+    |> put_description("nostrum")
+    |> put_url("https://google.com/")
+    |> put_timestamp("2016-05-05T21:04:13.203Z")
+    |> put_color(431_948)
+    |> add_field("Field 1", "Test")
+    |> add_field("Field 2", "More test", true)
+  ```
+
+  Alternatively, it is possible to build `Nostrum.Struct.Embed`s using standard map syntax.
+  However, we recommend sticking to the aforementioned "builder functions".
```
Comment (z64) [inline lib/nostrum/struct/embed.ex]: I don't really like leaving anecdotal comments like this unless you actually explain why, as the reader is going to be asking themselves this and left to guess exactly what you mean. I would expand on this or remove it.

File: lib/nostrum/struct/embed.ex [hunk 2]
```diff
@@ -87,7 +122,146 @@ defmodule Nostrum.Struct.Embed do
...
+    }
+
+    %__MODULE__{embed | author: author}
+  end
+
+  @doc ~S"""
+  Adds a `Nostrum.Struct.Embed.Field` under `:fields` in `embed`.
+  """
+  @spec add_field(t, Field.name(), Field.value(), Field.inline()) :: t
+  def add_field(embed, name, value, inline \\ nil)
```
Comment (z64) [inline lib/nostrum/struct/embed.ex]: I wonder about the verbiage here. Would a consistency of `put_field` be preferred? I know that internally this method functions "differently", but that is an implementation detail

**Comment by Kraigie** [comment]
Thanks!

**Pull request merged**